### DISCUSSION TITLE
Command `:Neorg journal today` throws error `Vim(Write):E212: Can't open file for writing: no such file or directory`

### DISCUSSION THREAD

**Discussion by piyushdivyankar1994** [nvim-neorg/neorg#725] [Q&A]
```
use {
    "nvim-neorg/neorg",
    config = function()
        require('neorg').setup {
            load = {
                ["core.defaults"] = {}, -- Loads default behaviour
                ["core.norg.concealer"] = {}, -- Adds pretty icons to your documents
                ["core.norg.dirman"] = { -- Manages Neorg workspaces
                    config = {
                        workspaces = {
                            notes = "~/notes",
                        },
                    },
                },
            },
        }
    end,
    run = ":Neorg sync-parsers",
    requires = "nvim-lua/plenary.nvim",
}
```

This is my config.

**Comment by dark-ether** [comment]
try again with the latest version, there was a bug where the journal toc wasn't properly generated if your current working directory was different than your journal directory.

**Reply by TheLeoP** [reply]
~~I'm having the same issue and the problem, I think, is that `vim.loop.fs_open` does not create a file nor its parent directories if the path doesn't exist on Windows. The following snippet of code (where `~/some/dir/a.norg` doesn't exist) fails with the error `ENOENT: no such file or directory C:\Users\user\some\dir\a.norg`~~

<strike>

```lua
local fname = "~/some/dir/a.norg"
fname = vim.fn.expand(fname, ":p")
print(fname)

local fd = vim.loop.fs_open(fname, "w", 438, function(err, fd)
	print(err)
	print(fd)
	print(fd)

	if fd then
		vim.loop.fs_close(fd)
	end
end)
```

</strike>

Edit: forget whay I just said, now I understand better how the code works. The problem, as you said, is the format using the date. It isn't considering the os separator, changing the code to something like the code bellow works as expected.

```lua
module.config.private = {
    strategies = {
        flat = "%Y-%m-%d.norg",
        nested = "%Y" .. neorg.configuration.pathsep .. "%m" .. neorg.configuration.pathsep .. "%d.norg",
    },
}
```

**Comment by alloc33** [comment]
does this plugin even work? `E212: Can't open file for writing: no such file or directory`

**Reply by max397574** [reply]
well I just checked and it works for me
